python/ray/tune/logger.py

```diff
@@ -358,8 +359,16 @@ def on_result(self, result):
                     full_attr, value, global_step=step)
             elif type(value) is list and len(value) > 0:
                 valid_result[full_attr] = value
-                self._file_writer.add_histogram(
-                    full_attr, value, global_step=step)
+                try:
+                    self._file_writer.add_histogram(
+                        full_attr, value, global_step=step)
+                # In case TensorboardX still doesn't think it's a valid value
+                # (e.g. `[[]]`), warn and move on.
+                except ValueError:
+                    logger.warning(
+                        "You are trying to log an invalid value ({}={}) "
+                        "via {}!".format(full_attr, value,
```

Comment:
Consider: `{} is not a valid histogram value. Histograms can only be created from numeric arrays.`

Reply:
{} would not even get into this block (the if checks for lists len>0). The case I stumbled upon was a list with another empty list in it. I know some structures are not valid, but it could be still useful to log this data e.g. in the result.json.
Maybe we actually should include all values in this elif block and then error/warn/log whatever value the users passed in.
The summary creation already failed at this point and I wanted to log the actual value (whatever format makes out of it) for debugging purposes.